src/Apps/Artist/routes.tsx

```diff
@@ -240,6 +240,22 @@ export const routes: RouteConfig[] = [
           ConsignRoute.preload()
         },
         displayFullPage: true,
+        render: ({ Component, props, match }) => {
+          if (!(Component && props)) {
+            return null
+          }
+
+          const artistPathName = match.location.pathname.replace("/consign", "")
+          const hasConsignPage = !!getConsignmentData(artistPathName)
+
+          // Check to see if artist is within top 20 consignable artists, and if
+          // not, redirect to home route.
+          if (!hasConsignPage) {
+            throw new RedirectException(artistPathName)
+          } else {
+            return <Component {...props} match={match} />
+          }
+        },
```

Comment:
What about something like this:

```ts
const noConsignmentData = !getConsignmentData(artistPathName)

if (noConsignmentData) {
  throw new RedirectException(artistPathName)
} else {
  return <Component {...props} match={match} />
}
```

Not a big deal, just suggesting that the name of the boolean condition reflect the case where we throw so that we don't have to recognize the exclamation points as much.

On the other hand, if we're grabbing this data here, we might as well pass it down to the component, no?

```ts
const consignmentData = getConsignmentData(artistPathName)

if (!consignmentData) {
  throw new RedirectException(artistPathName)
} else {
  return <Component {...props} match={match} consignmentData={consignmentData} />
}
```

This feels a little more clear to me, what do you think? If we have to check this data at the router level anyway seems like we should just pass it down rather than having it computed again.

Reply:
:+1: to the idea of passing the data into the component from here.

Reply:
Yup! both good suggestions, will do